python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -923,7 +923,10 @@ def _fetch_and_attach_row_count_metadata(
                     """,
                     fetch=True,
                 )
-            row_count = query_result[1][0]["row_count"]
+            query_result_table = query_result[1]
+            # some adapters do not output the column names, so we need
+            # to index by position
+            row_count = query_result_table[0][0]
```

Comment:
Was tested manually before it was changed to index-by-row-name, which evidently works in duckdb but not Snowflake unfortunately